File: refresh/src/main/java/com/mubai/refresh/activity/MainActivity.java
package com.mubai.refresh.activity;

import android.content.Context;
import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.BaseAdapter;
import android.widget.ListView;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import com.mubai.refresh.R;
import com.mubai.refresh.view.RefreshLayout;

import java.util.ArrayList;
import java.util.List;

/**
 * @author: lzw
 * //
 * @date: 2017/4/5 上午11:03
 * //
 * @desc:
 */

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "REFRESH";

    private RefreshLayout refreshLayout;

    private Handler handler = new Handler();

    private Runnable runnable = new Runnable() {
        @Override
        public void run() {
            Toast.makeText(MainActivity.this, "刷新完成了", Toast.LENGTH_SHORT).show();
            refreshLayout.setRefreshFinish();
        }
    };

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        refreshLayout = (RefreshLayout) findViewById(R.id.refresh_layout);
        refreshLayout.setRefreshListener(new RefreshLayout.RefreshListener() {
            @Override
            public void onRefresh() {
                handler.postDelayed(runnable, 4000);
            }
        });
    }

}

File: kotlin/src/main/java/com/lzw/kotlin/MainActivity.kt
package com.lzw.kotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import com.lzw.kotlin.adapter.MainRecyclerAdapter
import com.lzw.kotlin.data.Forecast
import com.lzw.kotlin.data.Student
import com.lzw.kotlin.data.User
import java.util.*

/**
 * //////////////////////////////////////////////////////////////////////////////
 * //
 * //      ┏┛ ┻━━━━━┛ ┻┓
 * //      ┃　　　　　　 ┃
 * //      ┃　　　━　　　┃
 * //      ┃　┗┛　  ┗┛　┃
 * //      ┃　　　　　　 ┃
 * //      ┃　　　┻　　　┃               @Author  lzw
 * //      ┃　　　　　　 ┃
 * //      ┗━┓　　　┏━━━┛               @Date  2017/2/7
 * //        ┃　　　┃   神兽保佑
 * //        ┃　　　┃   代码无BUG！      @Desc
 * //        ┃　　　┗━━━━━━━━━┓
 * //        ┃　　　　　　　    ┣━━━┛
 * //        ┃　　　　         ┏┛
 * //        ┗━┓ ┓ ┏━━━┳ ┓ ┏━┛
 * //          ┃ ┫ ┫   ┃ ┫ ┫
 * //          ┗━┻━┛   ┗━┻━┛
 * //
 * /////////////////////////////////////////////////////////////////////////////
 */
class MainActivity : AppCompatActivity() {
    private val items = listOf<String>(
            "A",
            "B",
            "C",
            "D",
            "E",
            "F",
            "G",
            "H",
            "I"
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val recyclerView = findViewById(R.id.recycler_view) as RecyclerView
        recyclerView.layoutManager = LinearLayoutManager(this)
        recyclerView.adapter = MainRecyclerAdapter(items)
        test("Example")
    }


    /**
     * 一切都是对象
     */
    fun test(S: String) {
        val a: Char = 'a'
        val b: Int = a.toInt()
        val c: Double = b.toDouble()
        val i = 1
        val d = 3.5
        val f = 2f
        val l = 1L

        //一个String可以像数组那样访问，并且被迭代
        val s = S
        val m = s[2] //:a
        To(this, m + "")
        for (c in s) {
            To(this, c + "")
        }

        val student = Student()
        student.height = "180"

        //修改Forecast其中一个属性
        val f1 = Forecast(Date(), 20f, "杭州")
        val f2 = f1.copy(temperature = 30f)
        //映射对象到变量中
        val (date, temperature, details) = f1

        //多声明
        val date2 = f1.component1()

        val temperature2 = f1.component2()

        val details2 = f1.component3()

    }
}
